course/test/src/isometry_TEST.cc

```diff
@@ -35,24 +36,37 @@ GTEST_TEST(Vector3Test, Vector3Operations) {
   EXPECT_EQ(p[1], 2.);
   EXPECT_EQ(p[2], 3.);
 
+
   std::stringstream ss;
   ss << p;
   EXPECT_EQ(ss.str(), "(x: 1, y: 2, z: 3)");
+  EXPECT_TRUE(Vector3::kUnitX == Vector3(1., 0., 0.));
+
+  EXPECT_TRUE(Vector3::kUnitX != Vector3(1., 1., 0.));
+  EXPECT_TRUE(Vector3::kUnitY == Vector3(0., 1., 0.));
 
-  EXPECT_TRUE(Vector3::kUnitX == Vector3(1., 0., 0));
-  EXPECT_TRUE(Vector3::kUnitX != {1., 1., 0});
-  EXPECT_TRUE(Vector3::kUnitY == {0., 1., 0});
+  //EXPECT_TRUE(Vector3::kUnitX != {1., 1., 0.});
+  //EXPECT_TRUE(Vector3::kUnitY == {0., 1., 0.});
+  
+  
   EXPECT_TRUE(Vector3::kUnitZ == Vector3::kUnitX.cross(Vector3::kUnitY));
   EXPECT_NEAR(Vector3::kUnitX.dot(Vector3::kUnitZ), 0., kTolerance);
 
   Vector3 t;
   EXPECT_EQ(t, Vector3::kZero);
-  t.x() = 1.;
+  //t.x() = 1.;
+  //t[1] = 2.;
+  //t.z() = 3.;
+  //EXPECT_EQ(t, p);
+
+  t[0] = 1.;
   t[1] = 2.;
-  t.z() = 3.;
+  t[2] = 3.;
```

Comment:
This was ok the way it was. You need to implement a `z()` overload that returns an lvalue reference so that you can use it on the left side of assingments.